cirq/work/observable_grouping_test.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2020 The Cirq developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cirq
+
+
+def test_group_settings_greedy():
```

Comment:
I'd love to see some more unit tests to document the detailed behavior for edge cases and simple cases as well, e.g. empty list, single items, single group, multi-group (that is the current one) situations.

Reply:
added a bunch of tests